### PR TITLE
Feature/add method fetch user contacts

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: tests/e2e/fetchMyContacts.test.ts [hunk 1]
```diff
@@ -1,172 +1,38 @@
...
 import { Wallet } from 'ethers';
-import { NULL_ADDRESS } from 'iexec/utils';
-import { WEB3_MAIL_DAPP_ADDRESS } from '../../dist/config/config';
-import { IExecWeb3mail, getWeb3Provider } from '../../dist/index';
-import { ValidationError, WorkflowError } from '../../dist/utils/errors';
-import { EnhancedWallet, IExec } from 'iexec';
+import { getWeb3Provider } from '../../dist/index';
+import { EnhancedWallet } from 'iexec';
+
+jest.unstable_mockModule('../../dist/web3mail/fetchUserContacts', () => ({
```
Comment (abbesBenayache) [inline tests/e2e/fetchMyContacts.test.ts:6]: JE te propose de ne pas modifier les tests e2e et de garder l'ancien code. 

En plus, les tests échoue => [Build](https://drone-product.iex.ec/iExecBlockchainComputing/web3mail-sdk/1321/1/4)

File: tests/e2e/fetchMyContacts.test.ts [hunk 2]
```diff
@@ -1,172 +1,38 @@
-import {
-  IExecDataProtector,
-  ProtectedDataWithSecretProps,
-  getWeb3Provider as dataprotectorGetWeb3Provider,
-} from '@iexec/dataprotector';
 import { beforeAll, describe, expect, it, jest } from '@jest/globals';
```
Comment (abbesBenayache) [inline tests/e2e/fetchMyContacts.test.ts:1]: JE te propose de ne pas modifier les tests e2e et de garder l'ancien code. 

En plus, les tests échoue => [Build](https://drone-product.iex.ec/iExecBlockchainComputing/web3mail-sdk/1321/1/4)

Comment (abbesBenayache) [inline tests/e2e/fetchMyContacts.test.ts:1]: il semble que les tests échouent actuellement ====>  [lien vers le build](https://drone-product.iex.ec/iExecBlockchainComputing/web3mail-sdk/1321/1/4).

File: tests/e2e/fetchMyContacts.test.ts (hunk 1 above)
Comment (abbesBenayache) [inline tests/e2e/fetchMyContacts.test.ts:6]: JE te propose de ne pas modifier les tests e2e et de garder l'ancien code. 

En plus, les tests échoue => [Build](https://drone-product.iex.ec/iExecBlockchainComputing/web3mail-sdk/1321/1/4)

File: tests/e2e/fetchUserContacts.test.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      .spyOn(web3mail, 'fetchUserContacts')
+      .mockRejectedValue(
+        new WorkflowError(
+          'Failed to fetch my contacts: wrong address is not a valid ethereum address',
+          new Error()
+        )
+      );
+
+    const expectedErrorMessage =
+      'Failed to fetch my contacts: wrong address is not a valid ethereum address';
```
Comment (abbesBenayache) [inline tests/e2e/fetchUserContacts.test.ts:105]:
```suggestion
    const expectedErrorMessage =
      'Failed to fetch user contacts: wrong address is not a valid ethereum address';
```

File: tests/e2e/fetchUserContacts.test.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+    expect(contacts).toEqual([]);
+  });
+
+  it('should throw a WorkflowError with a specific message', async () => {
+    jest
+      .spyOn(web3mail, 'fetchUserContacts')
+      .mockRejectedValue(
+        new WorkflowError(
+          'Failed to fetch my contacts: wrong address is not a valid ethereum address',
```
Comment (abbesBenayache) [inline tests/e2e/fetchUserContacts.test.ts:99]:
```suggestion
          'Failed to fetch user contacts: wrong address is not a valid ethereum address',
```

File: tests/e2e/fetchUserContacts.test.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
+import {
```
Comment (abbesBenayache) [inline tests/e2e/fetchUserContacts.test.ts:1]: il semble que les tests échouent actuellement ====>  [lien vers le build](https://drone-product.iex.ec/iExecBlockchainComputing/web3mail-sdk/1321/1/4).

File: src/web3mail/fetchUserContacts.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          WHITELIST_SMART_CONTRACT_ADDRESS.toLowerCase()
+      ) {
+        const contact = {
+          address: order.order.dataset.toLowerCase(),
+          owner: order.signer.toLowerCase(),
+          accessGrantTimestamp: order.publicationTimestamp,
+        };
+        myContacts.push(contact);
+      }
+    });
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply src/web3mail/fetchUserContacts.ts:69]: I don't want to change the initial implementation of the function. Moreover if i do that i need to add unit tests

File: tests/e2e/fetchUserContacts.test.ts (hunk 3 above)
Comment (Le-Caignec) [reply tests/e2e/fetchUserContacts.test.ts]: Commit: b9bd7449aadc3ef02bc22f9590897097571c2ac7

File: tests/e2e/fetchUserContacts.test.ts (hunk 4 above)
Comment (Le-Caignec) [reply tests/e2e/fetchUserContacts.test.ts]: Commit: b9bd7449aadc3ef02bc22f9590897097571c2ac7

File: tests/e2e/fetchUserContacts.test.ts (hunk 5 above)
Comment (Le-Caignec) [reply tests/e2e/fetchUserContacts.test.ts:1]: time out du to fetchMyContacts issue

File: tests/e2e/fetchMyContacts.test.ts (hunk 2 above)
Comment (Le-Caignec) [reply tests/e2e/fetchMyContacts.test.ts]: Commit: 14bea828191a37ed8d624a75d5572d7c75712512
The changes are due to format

File: tests/e2e/fetchMyContacts.test.ts (hunk 1 above)
Comment (Le-Caignec) [reply tests/e2e/fetchMyContacts.test.ts]: Commit: 14bea828191a37ed8d624a75d5572d7c75712512
The changes are due to format

File: tests/e2e/fetchMyContacts.test.ts (hunk 2 above)
Comment (Le-Caignec) [reply tests/e2e/fetchMyContacts.test.ts]: fix

**Review by abbesBenayache — APPROVED** [review]
👍 Everything looks good to me, but there are some unit tests not yet implemented.

**Pull request merged**